db/migrate/20231115163827_create_core_data_connector_works.rb

```diff
@@ -0,0 +1,10 @@
+class CreateCoreDataConnectorWorks < ActiveRecord::Migration[7.0]
+  def change
+    create_table :core_data_connector_works do |t|
+      t.references :project_model
+      t.jsonb :user_defined, default: {}
```

Comment:
Let's use the generator in the `user_defined_fields` gem to add the columns to the table. It also creates a database index to help performance.